### DISCUSSION TITLE
Problema con el CORS policy

### DISCUSSION THREAD

**Discussion by laurolmer** [Aparking/AparKing_Backend#75] [Q&A] [answered]
Buenas equipo, vengo a inaugurar el panel de Discussions a lo grande.
Os comento, @3ralon y yo mientras hacíamos el frontend de nuestras tareas, nos hemos encontrado con que el servidor de Angular lanza este error cuando tratamos de acceder a la ruta "http://localhost:8100/garages/1", para obtener los detalles del garaje con id 1. El error es el siguiente que os muestro en captura: 

![image](https://github.com/Aparking/AparKing_Backend/assets/73184962/b1a37377-e456-486d-8ee1-0febd7329285)

Por resumirlo brevemente, el problema es que debido a la política CORS Django no permite hacer llamadas cruzadas, entonces bloquea la petición de Angular. El caso es que buscando info en internet hay 2 soluciones, o metemos un proxy para "desviar" el error sin solucionarlo o se cambian las cabeceras, lo cual, ya está hecho en Django y aún así sigue bloqueando la llamada.

Os adjunto captura de cómo se ve en el backend y de cómo se ve en el frontend:
![image](https://github.com/Aparking/AparKing_Backend/assets/73184962/50b337d5-96fa-494a-a487-3921d83d4561)
![image](https://github.com/Aparking/AparKing_Backend/assets/73184962/9b7ec42f-f9e2-4b93-a8d6-f02c3eadf94c)

El settings.py de Django que tengo es:
![image](https://github.com/Aparking/AparKing_Backend/assets/73184962/6819ecf2-fad9-47a6-adf2-96e72c4b98a4)

**Comment by MarioArocaPaez** [comment]
Laura sé que parece una tontería, pero en una pregunta rápida a ChatGPT me ha dicho que es importante que en settings.py CorsMiddleware esté colocado por delante de CommonMiddleware, prueba a ponerlo a ver si por alguna casualidad soluciona algo o te da un mensaje distinto.

**Reply by laurolmer** [reply]
Que va, mira te muestro:
![image](https://github.com/Aparking/AparKing_Backend/assets/73184962/9da50fa8-c319-440b-b11b-ea004e697c96)

y esta es la respuesta del servidor:
![image](https://github.com/Aparking/AparKing_Backend/assets/73184962/7469608e-c6e6-41c0-9252-d1bc2657b24c)

**Comment by perealberto** [comment]
Buenas tardes compañeros, tras muchas horas chocándome contra la pantalla, he llegado a descubrir cuál puede ser el problema principal. El **NAVEGADOR**, cambiad a otro navegador. En mi caso, he cambiado a Chrome. Adjunto pruebas:

- Esta primera captura es utilizando el navegador Brave:
![image](https://github.com/Aparking/AparKing_Backend/assets/128732971/008714d7-20a3-45a3-ad5e-0231cba8415d)

- Esta segunda captura es utilizando el navegador Chrome:
![image](https://github.com/Aparking/AparKing_Backend/assets/128732971/099de5ff-218a-4400-bb9f-699f3b5d062c)

A esto hay que sumarle la configuración del CORS que hay que hacer en el backend. Algunos compañeros ya subieron los cambios, pero, por si acaso, adjunto el enlace donde se explica [cómo configurar el backend](https://pypi.org/project/django-cors-headers/).

**Reply by perealberto** [reply, marked as answer]
De acuerdo, tras volver a chocarme con la pared del CORS he dado con la solución definitiva. Resulta que el backend se despliega con la URL `http://127.0.0.1:8000`. Sin embargo, el archivo **[rest.service.ts](https://github.com/Aparking/AparKing_Frontend/blob/main/src/app/service/rest.service.ts)** del frontend hace peticiones a la URL `http://localhost:8000` que parece ser que no es la misma en cuanto a restricciones de este tipo en el navegador. Por tanto, si cambiamos la url del archivo **[rest.service.ts](https://github.com/Aparking/AparKing_Frontend/blob/main/src/app/service/rest.service.ts)** a 127.0.0.1, dejará de aparecer el problema.

La solución de este problema la he encontrado en el foro del canal de [discord oficial de Django](https://discord.com/invite/xcRH6mN4fa).

@Aparking/g1 @Aparking/g2 @Aparking/g3
